hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java

```diff
@@ -0,0 +1,397 @@
+      } catch (IOException ex) {
+        fail("testDoubleBufferWithMixOfTransactions failed");
+      }
+      Assert.assertNotNull(tableBucketInfo);
+
+      Assert.assertEquals(omBucketInfo.getVolumeName(),
+          tableBucketInfo.getVolumeName());
+      Assert.assertEquals(omBucketInfo.getBucketName(),
+          tableBucketInfo.getBucketName());
+      Assert.assertEquals(omBucketInfo.getCreationTime(),
+          tableBucketInfo.getCreationTime());
+    });
+  }
+
+  private void checkDeletedBuckets(Queue<OMBucketDeleteResponse>
+      deleteBucketQueue) {
+    deleteBucketQueue.forEach((omBucketDeleteResponse -> {
+      try {
+        Assert.assertNull(omMetadataManager.getBucketTable().get(
+            omMetadataManager.getBucketKey(
+                omBucketDeleteResponse.getVolumeName(),
+                omBucketDeleteResponse.getBucketName())));
+      } catch (IOException ex) {
+        fail("testDoubleBufferWithMixOfTransactions failed");
+      }
+    }));
+  }
+
+  public void testDoubleBuffer(int iterations, int bucketCount)
+      throws Exception {
+    try {
+      setup();
+      for (int i = 0; i < iterations; i++) {
+        Daemon d1 = new Daemon(() ->
+            doTransactions(UUID.randomUUID().toString(), bucketCount));
+        d1.start();
+      }
+
+      // We are doing +1 for volume transaction.
+      GenericTestUtils.waitFor(() ->
+              doubleBuffer.getFlushedTransactionCount() ==
+                  (bucketCount + 1) * iterations, 100,
+          120000);
+
+      Assert.assertTrue(omMetadataManager.countRowsInTable(
+          omMetadataManager.getVolumeTable()) == iterations);
+
+      Assert.assertTrue(omMetadataManager.countRowsInTable(
+          omMetadataManager.getBucketTable()) == (bucketCount) * iterations);
+
+      Assert.assertTrue(doubleBuffer.getFlushIterations() > 0);
+    } finally {
+      stop();
+    }
+  }
+
+
+  public void doTransactions(String volumeName, int buckets) {
+    doubleBuffer.add(createVolume(volumeName), trxId.incrementAndGet());
+    for (int i=0; i< buckets; i++) {
+      doubleBuffer.add(createBucket(volumeName, UUID.randomUUID().toString()),
+          trxId.incrementAndGet());
+      // For every 100 buckets creation adding 100ms delay
+
+      if (i % 100 == 0) {
+        try {
+          Thread.sleep(100);
+        } catch (Exception ex) {
+
+        }
+      }
+    }
+  }
+
+  private OMVolumeCreateResponse createVolume(String volumeName) {
+    OmVolumeArgs omVolumeArgs =
+        OmVolumeArgs.newBuilder()
+            .setAdminName(UUID.randomUUID().toString())
+            .setOwnerName(UUID.randomUUID().toString())
+            .setVolume(volumeName)
+            .setCreationTime(Time.now()).build();
+
+    VolumeList volumeList = VolumeList.newBuilder()
+        .addVolumeNames(volumeName).build();
+    return new OMVolumeCreateResponse(omVolumeArgs, volumeList);
+  }
+
+  private OMBucketCreateResponse createBucket(String volumeName,
```

Comment:
Let's move tests which depend on OM response classes to a separate test class.

Reply:
Done